src/core/args.c

```diff
@@ -409,25 +409,13 @@ void MVM_args_assert_nameds_used(MVMThreadContext *tc, MVMArgProcContext *ctx) {
     MVMuint16 i;
     if (size > 64) {
         for (i = 0; i < size; i++)
-            if (!ctx->named_used.byte_array[i]) {
-                char *c_param = MVM_string_utf8_encode_C_string(tc,
-                    ctx->args[ctx->num_pos + 2 * i].s);
-                char *waste[] = { c_param, NULL };
-                MVM_exception_throw_adhoc_free(tc, waste,
-                    "Unexpected named argument '%s' passed",
-                    c_param);
-            }
+            if (!ctx->named_used.byte_array[i])
+                MVM_args_throw_named_unused_error(tc, ctx->args[ctx->num_pos + 2 * i].s);
     }
     else {
         for (i = 0; i < size; i++)
-            if (!(ctx->named_used.bit_field & ((MVMuint64)1 << i))) {
-                char *c_param = MVM_string_utf8_encode_C_string(tc,
-                    ctx->args[ctx->num_pos + 2 * i].s);
-                char *waste[] = { c_param, NULL };
-                MVM_exception_throw_adhoc_free(tc, waste,
-                    "Unexpected named argument '%s' passed",
-                    c_param);
-            }
+            if (!(ctx->named_used.bit_field & ((MVMuint64)1 << i)))
+                MVM_args_throw_named_unused_error(tc, ctx->args[ctx->num_pos + 2 * i].s);
```

Comment:
In C, please use explicit brackets by default; as a side effect, that would've reduced the scope of the diff to contain only the block; which would've made it easier to review :smile:

Reply:
I'm generally a fan of explicit brackets, but the codebase has a **lot** of `if`s without them. I'd almost say without is more common. Should I change all the ones in this PR?

Reply:
What I meant to say is that, if you had used, or rather left, the brackets around the block, the diff would've contained only the actual contents of the block. Which makes it easy to assert that the conditional itself hadn't changed

Reply:
Oh yeah, I see now. Want me to add the braces back?